apis/python/tests/test_sparse_nd_array.py

```diff
@@ -388,3 +425,159 @@ def test_sparse_nd_array_reshape(tmp_path):
+        },
+        {
+            "coords": (3, 4),
+            "soma_dim_0": [3],
+            "soma_dim_1": [4],
+            "soma_data": [90304],
+        },
+        {
+            "coords": (slice(1, 2), slice(3, 4)),
+            "soma_dim_0": [1, 1, 2, 2],
+            "soma_dim_1": [3, 4, 3, 4],
+            "soma_data": [90103, 90104, 90203, 90204],
+        },
+        {
+            "coords": (slice(None), slice(3, 4)),
+            "soma_dim_0": [0, 0, 1, 1, 2, 2, 3, 3],
+            "soma_dim_1": [3, 4, 3, 4, 3, 4, 3, 4],
+            "soma_data": [90003, 90004, 90103, 90104, 90203, 90204, 90303, 90304],
+        },
+        {
+            "coords": (slice(1, 2), slice(None)),
+            "soma_dim_0": [1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2],
+            "soma_dim_1": [0, 1, 2, 3, 4, 5, 0, 1, 2, 3, 4, 5],
+            "soma_data": [
+                90100,
+                90101,
+                90102,
+                90103,
+                90104,
+                90105,
+                90200,
+                90201,
+                90202,
+                90203,
+                90204,
+                90205,
+            ],
+        },
+        {
+            "coords": (slice(None), slice(None)),
+            "soma_dim_0": [
+                0,
+                0,
+                0,
+                0,
+                0,
+                0,
+                1,
+                1,
+                1,
+                1,
+                1,
+                1,
+                2,
+                2,
+                2,
+                2,
+                2,
+                2,
+                3,
+                3,
+                3,
+                3,
+                3,
+                3,
+            ],
+            "soma_dim_1": [
+                0,
+                1,
+                2,
+                3,
+                4,
+                5,
+                0,
+                1,
+                2,
+                3,
+                4,
+                5,
+                0,
+                1,
+                2,
+                3,
+                4,
+                5,
+                0,
+                1,
+                2,
+                3,
+                4,
+                5,
+            ],
+            "soma_data": [
+                90000,
+                90001,
+                90002,
+                90003,
+                90004,
+                90005,
+                90100,
+                90101,
+                90102,
+                90103,
+                90104,
+                90105,
+                90200,
+                90201,
+                90202,
+                90203,
+                90204,
+                90205,
+                90300,
+                90301,
+                90302,
+                90303,
+                90304,
+                90305,
+            ],
+        },
+    ],
+)
+def test_sparse_nd_array_table_slicing(tmp_path, io):
```

Comment:
Comments on this test:
1. this needs to test ndim other than 2.   Ie, 1, 2, 3, ...
2. You could simplify this a lot if you programatically generated the data contents (eg, use numpy random) - that would radically simplify the parameterization.
3. There are legal coordinates that are not tested, eg, hybrid combinations of types, pa.chunked_array, pa.array, tuple, etc.   Example:      (1, slice(2)).